pyrival/data_structures/LinkedList.py

```diff
@@ -54,16 +54,26 @@ def __delitem__(self, index):
             self.__len -= 1
 
     def __repr__(self):
-        if not self:
-            return "{}()".format(self.__class__.__name__)
-        list_ = [self.__get_node(i).value for i in range(len(self))]
-        return "{}({})".format(self.__class__.__name__, list_)
+        return str(self.to_list())
+
+    def to_list(self):
+        elts = []
+        curr = self.sentinel.next
+        while curr != self.sentinel:
+            elts.append(curr.value)
+            curr = curr.next
+        return elts
 
     def append(self, value):
         sentinel = self.sentinel
         node = Node(value)
         self.insert_between(node, sentinel.prev, sentinel)
 
+    def appendleft(self, value):
+        sentinel = self.sentinel
+        node = Node(value)
+        self.insert_between(node, sentinel, sentinel.next)
```

Comment:
I know that having a local reference is faster in vanilla Python, but this doesn't really help in PyPy.

I'm not too sure about this, but I would prefer shorter code overall at this and similar instances. I don't use this code often enough, so if you disagree, I am happy to merge your PR as is.

Reply:
That was mostly copying from earlier the style of the initial file author, I can edit it out now.